lib/nnc/cmd/blas/mps/ccv_nnc_mul_mps.m

```diff
@@ -87,6 +87,171 @@ static int _ccv_nnc_mul_forw(const ccv_nnc_cmd_t cmd, const ccv_nnc_hint_t hint,
+			[inputShapedTypes addObject:mps_b1_shape];
+
+			MPSGraphTensor* mps_input_b2;
+			MPSGraphTensor* mps_b2 = ccv_nnc_mps_graph_tensor_input(graph, b2, b2->info.dim, b2->stride, &mps_input_b2);
+			[inputTensors addObject:mps_input_b2];
+			MPSGraphShapedType* mps_b2_shape = ccv_nnc_mps_graph_tensor_input_shape(b2, b2->info.dim, b2->stride);
+			[inputShapedTypes addObject:mps_b2_shape];
+			
+			if (g) {
+				MPSGraphTensor* mps_input_g;
+				MPSGraphTensor* mps_g = ccv_nnc_mps_graph_tensor_input(graph, g, g->info.dim, g->stride, &mps_input_g);
+				[inputTensors addObject:mps_input_g];
+				MPSGraphShapedType* mps_g_shape = ccv_nnc_mps_graph_tensor_input_shape(g, g->info.dim, g->stride);
+				[inputShapedTypes addObject:mps_g_shape];
+
+			MPSGraphTensor* mps_a = mps_g;
+			if (p != 1)
+			{
+				MPSGraphTensor* mps_p = [graph constantWithScalar:p dataType:ccv_nnc_mps_datatype(g->info.datatype)];
+				mps_a = [graph multiplicationWithPrimaryTensor:mps_a secondaryTensor:mps_p name:nil];
+
+			}
+			mps_a = [graph multiplicationWithPrimaryTensor:mps_a secondaryTensor:mps_b1 name:nil];
+
+			NSMutableArray<NSNumber*>* da_axes = [NSMutableArray new];
+			const int a_nd = ccv_nnc_tensor_nd(a->info.dim);
+			for (int i = 0; i < a_nd; i++) {
+				if (a->info.dim[i] != g->info.dim[i])
+					[da_axes addObject:@(i)];
+			}
+			mps_a = [graph reductionSumWithTensor:mps_a axes:da_axes name:nil];
+			[resultTensors addObject:mps_a];
+
+				if (h) {
+					MPSGraphTensor* mps_h = mps_g;
+					if (p != 1)
+					{
+						MPSGraphTensor* mps_p = [graph constantWithScalar:p dataType:ccv_nnc_mps_datatype(g->info.datatype)];
+						mps_h = [graph multiplicationWithPrimaryTensor:mps_h secondaryTensor:mps_p name:nil];
+					}
+					mps_h = [graph multiplicationWithPrimaryTensor:mps_h secondaryTensor:mps_b2 name:nil];
+
+					NSMutableArray<NSNumber*>* dh_axes = [NSMutableArray new];	
+					const int h_nd = ccv_nnc_tensor_nd(h->info.dim);
+					for (int i = 0; i < h_nd; i++) {
+						if (h->info.dim[i] != g->info.dim[i])
+							[dh_axes addObject:@(i)];
+					}
+					mps_h = [graph reductionSumWithTensor:mps_h axes:dh_axes name:nil];
+					[resultTensors addObject:mps_h];
+				}
+				
+			} else {
+				// empty mps_g for target broadcast shape with 1.0 for each elements	 
+				MPSGraphTensor* mps_g = [graph constantWithScalar:1.0 shape:mps_g_shape dataType:ccv_nnc_mps_datatype(a->info.datatype)];
+				MPSGraphTensor* mps_a = mps_g;
+				if (p != 1)
+				{
+					MPSGraphTensor* mps_p = [graph constantWithScalar:p dataType:ccv_nnc_mps_datatype(b1->info.datatype)];
+					mps_a = [graph multiplicationWithPrimaryTensor:mps_a secondaryTensor:mps_p name:nil];
+				}
+				mps_a = [graph multiplicationWithPrimaryTensor:mps_a secondaryTensor:mps_b1 name:nil];
+				NSMutableArray<NSNumber*>* da_axes = [NSMutableArray new];	
+				const int a_nd = ccv_nnc_tensor_nd(a->info.dim);
+				for (int i = 0; i < a_nd; i++) {
+					if (a->info.dim[i] != mps_g_shape[i].integerValue)
```

Comment:
I am using mps_g_shape as a workaround to access gdim inside executable block... 
may need a review to double check I did this in the right way...

Reply:
The simple way to workaround this is to have an alias before this block:
```
const int* gdim_a = gdim;
```
This way, compiler won't complain that block cannot capture static array.